# Differential gene expression and gene ontology enrichment analysis
Expression levels were determined for each gene using the entire sequencing read (rather than only the 3â€™ end, as typical when determining Pol II location from NET-seq data) and normalized using DESeq2 (Love et al., 2014). Differential expression analysis between deletion strains (with replicates) and wild-type strains was performed using DESeq2 (Love et al., 2014) for all UCSC known genes. Genes were considered differentially expressed if they had an adjusted p-value < 0.05 and an absolute log2 fold change > 1.0. GO term enrichment analysis was performed using the PANTHER GO Enrichment Analysis (Ashburner et al., 2000; Mi et al., 2019; The Gene Ontology Consortium, 2019; Thomas et al., 2003) and enriched terms for biological processes, molecular function, and cellular components were recorded for up-regulated differentially expressed genes, down-regulated differentially expressed genes, and all differentially expressed genes. 

# 
